WalletWasabi/WabiSabi/Models/DecompositionAlgs/RestrictedSetDecomposer.cs

```diff
@@ -0,0 +1,56 @@
+using NBitcoin;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class RestrictedSetDecomposer
```

Comment:
I think we should just call it `GreedyDecomposer` since that's the strategy it uses, and we can extract the more abstract `RestrictedSetDecomposer` by refactoring as necessary as a later point.

Reply:
ACK